From a8662752108579ed0bc1fa1566b1a380fcf825f0 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Wed, 1 May 2019 19:53:17 +0200
Subject: [PATCH 306/345] ATA: Disable soft reset for ASM1092 sata port
 multiplier

Soft reset break sata detection and disk gets mis-classified.

Signed-off-by: Sergey Temerkhanov <stemerkhanov@cavium.com>
Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/ata/libata-pmp.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/ata/libata-pmp.c b/drivers/ata/libata-pmp.c
index 85aa76116a30..be14aee56439 100644
--- a/drivers/ata/libata-pmp.c
+++ b/drivers/ata/libata-pmp.c
@@ -467,6 +467,14 @@ static void sata_pmp_quirks(struct ata_port *ap)
 			if (link->pmp == 4)
 				link->flags |= ATA_LFLAG_DISABLED;
 		}
+	} else if (vendor == 0x1b21 && devid == 0x1092) {
+		ata_for_each_link(link, ap, EDGE) {
+			/* SRST breaks detection and disks get misclassified
+			* LPM disabled to avoid potential problems
+			*/
+			link->flags |= ATA_LFLAG_NO_SRST |
+				       ATA_LFLAG_NO_LPM;
+		}
 	}
 }
 
-- 
2.25.1

